components/feature/pwa/src/main/java/mozilla/components/feature/pwa/WebAppShortcutManager.kt

```diff
@@ -9,17 +9,42 @@ import android.content.Intent
 import android.content.pm.ShortcutManager
 import android.os.Build.VERSION.SDK_INT
 import android.os.Build.VERSION_CODES
+import androidx.annotation.VisibleForTesting
 import androidx.core.content.getSystemService
 import androidx.core.content.pm.ShortcutInfoCompat
 import androidx.core.content.pm.ShortcutManagerCompat
 import androidx.core.graphics.drawable.IconCompat
 import androidx.core.net.toUri
+import mozilla.components.browser.icons.BrowserIcons
+import mozilla.components.browser.icons.decoder.AndroidIconDecoder
+import mozilla.components.browser.icons.decoder.ICOIconDecoder
+import mozilla.components.browser.icons.extension.toIconRequest
+import mozilla.components.browser.icons.generator.DefaultIconGenerator
+import mozilla.components.browser.icons.loader.DataUriIconLoader
+import mozilla.components.browser.icons.loader.DiskIconLoader
+import mozilla.components.browser.icons.loader.HttpIconLoader
+import mozilla.components.browser.icons.loader.MemoryIconLoader
+import mozilla.components.browser.icons.preparer.DiskIconPreparer
+import mozilla.components.browser.icons.preparer.MemoryIconPreparer
+import mozilla.components.browser.icons.preparer.TippyTopIconPreparer
+import mozilla.components.browser.icons.processor.AdaptiveIconProcessor
+import mozilla.components.browser.icons.processor.ColorProcessor
+import mozilla.components.browser.icons.processor.DiskIconProcessor
+import mozilla.components.browser.icons.processor.MemoryIconProcessor
+import mozilla.components.browser.icons.processor.ResizingProcessor
+import mozilla.components.browser.icons.utils.IconDiskCache
+import mozilla.components.browser.icons.utils.IconMemoryCache
 import mozilla.components.browser.session.Session
 import mozilla.components.concept.engine.manifest.WebAppManifest
+import mozilla.components.concept.fetch.Client
 import mozilla.components.feature.pwa.ext.installableManifest
 
+private val pwaIconMemoryCache = IconMemoryCache()
+private val pwaIconDiskCache = IconDiskCache()
+
 class WebAppShortcutManager(
     private val storage: ManifestStorage,
+    private val icons: BrowserIcons,
```

Comment:
I think we should construct this internally and only ask for context/client. Otherwise there's a high chance that you get a wrongly configured `BrowserIcons` instance - even though you want exactly what comes out of `webAppIcons()`.